### PR TITLE
Extract data

### PR DESCRIPTION
**Pull request by dwhenry**

### CODE REVIEW

File: app/services/booking_bug_connection.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require 'faraday'
+require 'faraday_middleware'
+
+class BookingBugConnection
```
Comment (andrewgarner) [inline app/services/booking_bug_connection.rb]: Is this a service?

File: .env [hunk 2]
```diff
@@ -0,0 +1,6 @@
+BOOKING_BUG_ENVIRONMENT=treasurydev
+BOOKING_BUG_COMPANY_ID=37004
+BOOKING_BUG_API_KEY=1962b5a78fca8b79351e969c5832bf60
+BOOKING_BUG_APP_ID=fd0cd097
+BOOKING_BUG_EMAIL=developers@pensionwise.gov.uk
+BOOKING_BUG_PASSWORD=zZuNFWRD9IqD
```
Comment (andrewgarner) [inline .env]: This should not be public.

File: config/initializers/booking_bug_config.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+module Config
```
Comment (andrewgarner) [inline config/initializers/booking_bug_config.rb]: `Configurable` might make more sense if this is a concern that can be mixed in.

File: config/initializers/booking_bug_config.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      yield(self) if block_given?
+    end
+
+    def attributes
+      self.class.attrs
+    end
+  end
+end
+
+BookingBug::ATTRIBUTES = %w(environment company_id api_key app_id email password).freeze
```
Comment (andrewgarner) [inline config/initializers/booking_bug_config.rb]: Why are we telling `BookingBug` how it should define it's configuration? Isn't it `BookingBug`'s responsibility to define it's interface for configuration?

File: features/booking/extract_data.feature [hunk 5]
```diff
@@ -17,6 +17,26 @@ Feature: Extract booking data
   Scenario: Isolating the changed data to allow selective processing
     When the booking data is a mix of new and already extracted data
     Then only the changed booking data is extracted
+  
+  @wip
```
Comment (andrewgarner) [inline features/booking/extract_data.feature]: These are duplicates of scenarios we already have.

File: features/support/env.rb [hunk 6]
```diff
@@ -1,3 +1,57 @@
...
+# exceptions and display an error page (just like when the app is running in production).
+# Typical scenarios where you want to do this is when you test your error pages.
+# There are two ways to allow Rails to rescue exceptions:
+#
+# 1) Tag your scenario (or feature) with @allow-rescue
+#
+# 2) Set the value below to true. Beware that doing this globally is not
+# recommended as it will mask a lot of errors for you!
+#
+ActionController::Base.allow_rescue = false
```
Comment (andrewgarner) [inline features/support/env.rb]: We don't have any controllers.

File: features/support/env.rb [hunk 7]
```diff
@@ -1,3 +1,57 @@
...
+#
+#   Before('~@no-txn', '~@selenium', '~@culerity', '~@celerity', '~@javascript') do
+#     DatabaseCleaner.strategy = :transaction
+#   end
+#
+
+# Possible values are :truncation and :transaction
+# The :transaction strategy is faster, but might give you threading problems.
+# See https://github.com/cucumber/cucumber-rails/blob/master/features/choose_javascript_database_strategy.feature
+Cucumber::Rails::Database.javascript_strategy = :truncation
```
Comment (andrewgarner) [inline features/support/env.rb]: We don't have any javascript.

File: features/booking/extract_data.feature (hunk 5 above)
Comment (dwhenry) [inline features/booking/extract_data.feature]: This is outside the scope of this pull request as I am trying to focus on data extraction only. 

I plan to do a review of the features as I implement them, and then do a full review with @amansinghb once all the code bits I have been working on have been merged.

File: app/services/booking_bug_connection.rb (hunk 1 above)
Comment (dwhenry) [inline app/services/booking_bug_connection.rb]: It is a component that at this stage is used by the service, it might be better called BookingBug::Connnection but that did not feel right. 

I think this is as good a place as any for this object at the moment. Once we add a second connection object we should think more about the best location for it. 

Happy to move if you have strong feelings, otherwise I will mark this for future consideration.
